Add new chapter, on publication pipeline with resources (#77)
esciencecenter-digital-skills/software-support-essentials: static/stories/publication.md

:::Chapter{headline="Intro slides" presentation="slides-publication.md"}

:::

:::Chapter{headline="Software publication pipeline" widemd=1}

## Steps
It is recommended that all steps in the pipeline are followed, but they are not dependent on eachother. In this pipeline we provide resources to achieve the whole pipeline using GitHub or GitLab for hosting the source code 
* Software repository
* Tag/semantic versioning
* Get a persistent identifier
* Registries
* Software Paper

## Software repository
If possible, during the development of Research Software, changes should be tracked using dedicated version control software like [git](https://git-scm.com/).  

> Put all source files in a public version-controlled repository, preferably GitHub or GitLab. Both GitHub and Gitlab provide online repositories for hosting source code. The nice thing about GitHub and GitLab is that it also stores the history of changes to the source code. As can be inferred from their names, they were designed to be used together with Git version control.  

Reviewer: > nice thing about GitHub and GitLab is that it also stores the history of changes

This is more a feature of git itself, so maybe we can reword this

> Put all source files in a public version-controlled repository, preferably GitHub or GitLab. Both GitHub and Gitlab provide online repositories for hosting source code. The nice thing about GitHub and GitLab is that it also stores the history of changes to the source code. As can be inferred from their names, they were designed to be used together with Git version control.  

Author: ```suggestion
Put all source files in a public version-controlled repository, preferably GitHub or GitLab. Both GitHub and Gitlab provide online repositories for hosting source code. As can be inferred from their names, they were designed to be used together with Git version control.  The nice thing about using git together with GitHub and GitLab is that it stores the complete history of changes to the source code, rather than just a single version . 
```

A quickstart tutorial for...

* [GitHub](https://docs.github.com/en/get-started/quickstart/create-a-repo)
* [GitLab](https://docs.gitlab.com/ee/user/project/index.html)

## Versions, tags and releases

> Every time software changes, and that history is stored, that can be considered a new **version** of the software. It is useful to add special identifiers, called **tags** to new versions that mark significant changes, so that specific versions of your software can be cited or referenced as dependency by other software. If you use git for version control, [creating tags in git](https://git-scm.com/book/en/v2/Git-Basics-Tagging)

Reviewer: ```suggestion
Every time software changes, and that history is stored, that can be considered a new **version** of the software. It is useful to add special identifiers, called **tags** to new versions that mark significant changes, so that specific versions of your software can be cited or referenced as dependency by other software. If you use git for version control, you can [create tags in git](https://git-scm.com/book/en/v2/Git-Basics-Tagging)
```

> It is good practice to use **[semantic versioning](https://semver.org/)** to tag the software, in this way, information about the impact of the change from one version to the next is stored in the tag. Usually, tagged versions of the software are opened up to the public in a **software release**, together with updated documentation and a documentation of the changes since the previous release in a so called **changelog**.

Reviewer: ```suggestion
It is good practice to use **[semantic versioning](https://semver.org/)** to tag the software. In this way, information about the impact of the change from one version to the next is stored in the tag. Usually, tagged versions of the software are opened up to the public in a **software release**, together with updated documentation and a documentation of the changes since the previous release in a so called **changelog**.
```

Reviewer: Also, maybe "associated to the tag" (rather than "stored in the tag")? Or "stored in the tagged commit"? The latter sounds too technical though without going into what git is and does

The documentation of creating releases... 

* [on GitHub](https://docs.github.com/en/repositories/releasing-projects-on-github)
* [on GitLab](https://docs.gitlab.com/ee/user/project/releases/)

## Get a persistent identifier
Specific versions of the software should be published to an archiving service that provides persistent identifiers

> For example, Zenodo is a long term archiving service for scientific data and software.

Reviewer: Zenodo should have a hyperlink or it should be in the references

> For example, Zenodo is a long term archiving service for scientific data and software.

Author: ```suggestion
For example, [Zenodo](https://zenodo.org/) is a long term archiving service for scientific data and software.
```
